Documentation/Contributing.md

```diff
@@ -0,0 +1,160 @@
+```
+2. Run your first build:
+```
+$> cd swift-package-manager
+$> swift build
+```
+If the build process ends with exit code 0, the build is successful (we have an Enhancement Radar to implement a message for successful build and a short output on where the generated binaries are: rdar://69970428).  
+After a successful build (currently), you should see something like this:
+```
+[476/476] Linking swift-package
+```
+A `.build/` folder will be generated and it should have inside a similar structure (including build binaries):
+```
+artifacts/
+checkouts/
+debug/
+repositories
+x86_64-apple-macosx
+debug.yaml
+manifest.db
+workspace-state.json 
+```
+Binaries (in the example above) are in `x86_64-apple-macosx/`.  
+If you need to build the generated binaries, run `swift-build` in inside `.build/`:
+```
+./.build/x86_64-apple-macosx/debug/swift-build
+```
+
+### Code Editor
+If you are contributing using macOS, the best option is to use Xcode to build and run test SwiftPM. 
+
+### Troubleshooting
+* If during `swift build` you encounter these outputs:
+```
+/../swift-package-manager/Sources/SPMTestSupport/misc.swift:93:35: warning: parameter 'file' with default argument '#file' passed to parameter 'file', whose default argument is '#filePath'
+        XCTFail("\(error)", file: file, line: line)
+                                  ^
+/../swift-package-manager/Sources/SPMTestSupport/misc.swift:37:26: note: did you mean for parameter 'file' to default to '#filePath'?
+    file: StaticString = #file,
+                         ^~~~~
+                         #filePath
+/../swift-package-manager/Sources/SPMTestSupport/misc.swift:93:35: note: add parentheses to silence this warning
+        XCTFail("\(error)", file: file, line: line)
+                                  ^
+                                  (   )
+```
+Do not worry, since those are known warnings that will be addressed at some point.  
+Warnings differ depending on the platform and they can be seen from time to time due the amount of contributions.  
+Our goal is to constantly monitor warnings and work on fix them (even if they are not affecting a successful implementation).
+* If during `swift build` you encounter this error:
+```
+/../apple-repos/swift-package-manager/.build/checkouts/swift-driver/Sources/SwiftDriver/Explicit Module Builds/InterModuleDependencyGraph.swift:102:3: error: unknown attribute '_spi'
+  @_spi(Testing) public var isFramework: Bool
+  ^
+```
+Make sure you are using SwiftPM 5.3
+```
+$> swift package --version
+Swift Package Manager - Swift 5.3.0
+```
+* If during `swift build` you encounter this error:
+```
+/../swift-package-manager/Sources/PackageLoading/Target+PkgConfig.swift:84:36: error: type 'PkgConfigError' has no member 'prohibitedFlags'
+            error = PkgConfigError.prohibitedFlags(filtered.unallowed.joined(separator: ", "))
+                    ~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~
+```
+Make sure to update your TSC (Tools Support Core):
```

Comment:
hmm TSC is a dependency so should be fetched properly, so not sure about this one

Reply:
~~I guess we could say run "swift package update"~~